channel/buffs.go

```diff
@@ -539,6 +539,97 @@ func (cb *CharacterBuffs) AddItemBuffFromCC(itemID int32, expiresAtMs int64) {
 	}
 }
 
+// AddMobDebuff applies a debuff from a mob skill to the player
+func (cb *CharacterBuffs) AddMobDebuff(skillID, level byte, durationSec int16) {
+	if cb.plr == nil {
+		return
+	}
+
+	// Map mob skill IDs to buff bit positions and determine debuff values
+	var bits []int
+	var debuffValues map[int]int16 = make(map[int]int16)
+	
+	switch skillID {
+	case skill.Mob.Seal:
+		bits = []int{BuffSeal}
+		debuffValues[BuffSeal] = 1
+	case skill.Mob.Darkness:
+		bits = []int{BuffDarkness}
+		debuffValues[BuffDarkness] = 1
+	case skill.Mob.Weakness:
+		bits = []int{BuffWeakness}
+		debuffValues[BuffWeakness] = 1
+	case skill.Mob.Stun:
+		bits = []int{BuffStun}
+		debuffValues[BuffStun] = 1
+	case skill.Mob.Curse:
+		bits = []int{BuffCurse}
+		debuffValues[BuffCurse] = 1
+	case skill.Mob.Poison:
+		bits = []int{BuffPoison}
+		debuffValues[BuffPoison] = 1
+	case skill.Mob.Slow:
+		// Slow reduces speed - send negative value
+		bits = []int{BuffSpeed}
+		debuffValues[BuffSpeed] = -int16(level * 10) // Reduce speed by level * 10
+	default:
+		return
+	}
+
+	if len(bits) == 0 {
+		return
+	}
+
+	// Build mask bytes
+	maskBytes := buildMaskBytes64(bits)
+	
+	// Encode mob skill as negative to distinguish from player skills
+	mobSkillID := -int32(skillID)
+	
+	// Build value triples for the debuff
+	out := make([]byte, 0, 16)
+	for _, bit := range bits {
+		// Get the debuff value for this bit
+		val := debuffValues[bit]
+		out = append(out, byte(val), byte(val>>8))
+		
+		// Source ID (negative mob skill ID)
+		out = append(out, byte(mobSkillID), byte(mobSkillID>>8), byte(mobSkillID>>16), byte(mobSkillID>>24))
+		
+		// Duration in seconds
+		out = append(out, byte(durationSec), byte(durationSec>>8))
+	}
+
+	// Send to self
+	const extra byte = 0
+	const delay int16 = 0
+	cb.plr.Send(packetPlayerGiveBuff(maskBytes, out, delay, extra))
```

Comment:
This is causing the player to crash. It seems the logic behind this isn't properly being constructed.